### ISSUE TITLE
generate bitcoin address from pubkey

### ISSUE THREAD

**Issue by ja-heller** [Bit-Wasp/bitcoin-php#756]
I'm trying to get the bitcoin address from a given pubkey.

Using this code:

```
        $pubkey = PublicKeyFactory::fromHex("034ea055e86c6a18ed05e2d58ba0298040319d216f35c3c1963e88e4826e05a9fd");
        die(var_dump($pubkey->getAddress()));
```

I've used [https://bitcoin.stackexchange.com/questions/48023/how-to-generate-addresses-in-php](this post) regarding how to do so.
Now I get:

> Call to undefined method BitWasp\Bitcoin\Crypto\EcAdapter\Impl\PhpEcc\Key\PublicKey::getAddress()

Has the method name changed or is the way I'm trying to get the address of the pubkey wrong in general?

**Comment by afk11** [comment]
Yep, it's written about an old version. We removed getAddress from there because it coupled peoples applications with the preferences of the developers here, and not everyone wants their scripts changing randomly.

this one still works:
`new PayToPubKeyHashAddress($pubKey->getPubKeyHash())`

no idea what version you're using, but check what version of the library you have and look for relevant examples on github for that version/

**Comment by ja-heller** [comment]
@afk11 

Thanks for the fast reply.
Using the code suggestion above I get:

```
PayToPubKeyHashAddress {#254 ▼
  #hash: Buffer {#255 ▼
    +"size": 20
    +"buffer": "0xf364cef85350e18b4055d0b43dacc3bcbb68e0eb"
  }
}
```

What's the best way to get a "human readable" address now?
